pageserver/src/tenant.rs

```diff
@@ -2570,6 +2594,14 @@ impl Tenant {
             cached_logical_sizes: tokio::sync::Mutex::new(HashMap::new()),
             cached_synthetic_tenant_size: Arc::new(AtomicU64::new(0)),
             eviction_task_tenant_state: tokio::sync::Mutex::new(EvictionTaskTenantState::default()),
+            compaction_circuit_breaker: std::sync::Mutex::new(CircuitBreaker::new(
+                format!("compaction-{tenant_shard_id}"),
+                5,
+                // Compaction can be a very expensive operation, and might leak disk space.  It also ought
+                // to be infallible, as long as remote storage is available.  So if it repeatedly fails,
+                // use an extremely long backoff.
+                Some(Duration::from_secs(3600 * 24)),
```

Comment:
make it configurable in the future?

Reply:
Maybe, I can also imagine wanting an http to reset a circuit breaker at runtime.  We will probably have more than one of these, so need to think about whether we want/need to expose configs for them individually, it could get quite noisy.